langkit/expressions/base.py

```diff
@@ -3190,6 +3191,10 @@ def __init__(self, expr, prefix, name=None, doc=None, public=None,
 
         :param bool dump_ir: If true, dump the tree of resolved expressions for
             this property.
+
+        :param lazy_field: Whether the goal of this property is to initialize a
```

Comment:
In my mind, a lazy field would be more similar to a user/parsing field for users than to a property (it cannot have arguments, evaluates once), and I thought it was for this reason that we called them lazy *fields*. Maybe we should discuss this live?

Reply:
It's also more similar to a memoized property, insofar as it is evaluated at the moment it is accessed, not when it is created. The barrier between the two is relatively blurry, and that's not the reason why I think you should reformulate: 

In the current DSL, the property **represents** the field in the source. There is no two different entities for the client (the field and the property), but only one (the field/property). So making the distinction only serves to confuse users.

Reply:
FTR as far as I can tell that's also what you're telling in the source when you write:

```py
@lazy_field
def pouet():
   return 12
```

and not

```py
@initializes_lazy_field(name="pouet")
def pouet():
   return 12
```

in the second example we can see well what it would be like if the "initializing property" and the "lazy field" were two different things

Reply:
As discussed live, this docstring is for Langkit’s internals: users aren’t supposed to use this constructor nor have access to this argument, so we can leave it this way.